sympy/matrices/expressions/matexpr.py

```diff
@@ -110,7 +110,11 @@ def __rmul__(self, other):
     def __pow__(self, other):
         if not self.is_square:
             raise ShapeError("Power of non-square matrix %s" % self)
-        if other is S.NegativeOne:
+        if self.is_ZeroMatrix:
+            return self
```

Comment:
@jksuom  That is because, zero matrix raised to any positive power is zero matrix itself, whereas 0th matrix raised to power 0 is an Identity matrix.